demux/utils/samplesheet.py

```diff
@@ -206,6 +208,65 @@ def _validate_uniq_index(samplesheet):
         return True
 
 
+class MiSeqSamplesheet(Samplesheet):
+
+    def __init__(self, samplesheet_path):
+        self.samplesheet_path = samplesheet_path
+        converted_sheet_path = self.convert_sheet_type(samplesheet_path)
+        self.parse(converted_sheet_path)
+
+    def convert_sheet_type(self, samplesheet_path):
+        """
+        Conver miseq to hiseq style samplesheet for demultiplexing.
+        """
+        dirname = os.path.dirname(samplesheet_path)
+        flowcell_id = dirname.split(os.sep)[-1].split('_')[-1] 
+        deplex_file = os.path.join(dirname, 'demultiplex_sheet.csv')
+        header_line = "FCID,Lane,SampleID,SampleRef,Index,Description,Control,Recipe,Operator,SampleProject\n"
+        forbidden_name_chars = [' ','-', '/']
+
+        with open(samplesheet_path, 'r') as inputfile open(deplex_file 'w') as outputfile:
+            outputfile.write(header_line)
```

Comment:
Writing to disk shouldn't be handled in this class. The idea is to parse a samplesheet, perform some operations on it, and return the samplesheet in some form or another (e.g. only the `[Data]` part in `self.lines()` or `to_demux()` which returns a demux-ready samplesheet as a printable string based on what it got in.